src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -151,60 +152,49 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
 
             await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            _componentModel = (IComponentModel)await GetServiceAsync(typeof(SComponentModel)).ConfigureAwait(true);
-            cancellationToken.ThrowIfCancellationRequested();
-            Assumes.Present(_componentModel);
-
-            // Ensure the options persisters are loaded since we have to fetch options from the shell
-            LoadOptionPersistersAsync(_componentModel, cancellationToken).Forget();
-
-            _workspace = _componentModel.GetService<VisualStudioWorkspace>();
-
             // Fetch the session synchronously on the UI thread; if this doesn't happen before we try using this on
             // the background thread then we will experience hangs like we see in this bug:
             // https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems?_a=edit&id=190808 or
             // https://devdiv.visualstudio.com/DevDiv/_workitems?id=296981&_a=edit
-            var telemetryService = (VisualStudioWorkspaceTelemetryService)_workspace.Services.GetRequiredService<IWorkspaceTelemetryService>();
-            telemetryService.InitializeTelemetrySession(TelemetryService.DefaultSession);
```

Comment:
❔ What if we just make `FileLogger._enabled` lazy?

Reply:
That would be possible but I wanted to avoid dependency on that.